frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -86,6 +112,11 @@ function FileManagerHistory:onMenuHold(item)
                         ok_callback = function()
                             filemanagerutil.purgeSettings(item.file)
                             require("readhistory"):fileSettingsPurged(item.file)
+                            if self.filter ~= "all" then
+                                self._manager:fetchStatuses(false)
+                            else
+                                self.statuses_fetched = false
+                            end
```

Comment:
Just tested this "Reset settings", with History without filter and not having invoked the top left button, and when taping on the `Reset` button, I get tons of log `DEBUG DocSettings: data is read from` (so, actually reading the docsettings, while it's not needed when no filter), which I don't get without this PR.

Reply:
Oh yes, the wrong `self`.

Reply:
The wrong `self`?

Reply:
That should fix it. I still don't get why some callbacks use `self` and others use `self._manager`

Reply:
Yep, seems fixed.
As about self vs self._manager, it took me a few minutes to get why:
https://github.com/koreader/koreader/blob/4f3d86708e846d564e7ddf77493ea69f8de9e3b3/frontend/apps/filemanager/filemanagerhistory.lua#L190-L203
and how we use/set `onMenuHold` and `onSetRotationMode` (while the `onLeftButtonTap` feels immune to this confusion).
We define `FileManagerHistory:onMenuHold(item)` as a method (`:`), which makes the first argument implicitely be self, it's the same as `FileManagerHistory.onMenuHold(self, item)`, a function/callable with 2 arguments.
We then pass it when creating a Menu instance, and it overrides the class-defined method/callable `Menu:onMenuHold`.
When Menu calls it as `Menu:onMenuHold`, it's like it is called as  `Menu.onMenuHold(self,item)` with self here being the Menu instance.
Luckily, when we created the Menu instance, we pass `_manager = self = the FileManagerHistory instance` that we can hopefully reach from the callback we defined.

Dunno if this way of doing it was a necessity, or we just inherited it some clumsy bit of code.
Feels just using `onMenuHold = function(item) self:onMenuHold(item) end` would be the right way to handle this. But no idea if this would cause other issues.

Reply:
> Feels just using `onMenuHold = function(item) self:onMenuHold(item) end` would be the right way to handle this.

Unfortunately this doesn't work, this way `onMenuHold` cannot get correct `item`.
If we want to use `self` instead of `self._manager` in `onMenuHold`, this works:
```Lua
local hist = self
function self.hist_menu:onMenuHold(item)
    hist:onMenuHold(item)
end

```
(We have similar trick in bookmarks)

Reply:
> this way `onMenuHold` cannot get correct `item`.

"Incorrect item" doesn't mean a wrong long-pressed book from the history.
It is an item from some other table (something very long, maybe the full Menu table).

Reply:
Yeah, that was probably needed, to access both Menu and FileManagerHistory instances' own stuff - so, one way or the other...
I'm not suggesting we should rewrite it, as it works.
But the person who wrote it that way would have saved us some time if he had just put some comments explaining why it's needed to do it that way and what is what.

Reply:
> > Feels just using `onMenuHold = function(item) self:onMenuHold(item) end` would be the right way to handle this.
> 
> Unfortunately this doesn't work, this way `onMenuHold` cannot get correct `item`. If we want to use `self` instead of `self._manager` in `onMenuHold`, this works:
> 
> ```lua
> local hist = self
> function self.hist_menu:onMenuHold(item)
>     hist:onMenuHold(item)
> end
> ```
> 
> (We have similar trick in bookmarks)

What about `onMenuHold = function(_, item) self:onMenuHold(item) end`?

(First arg is the object when it's used as a method, which is probably why you were seeing a MenuThing in item)

Reply:
Great, seems that works! I'll test this way in some more cases.

Reply:
I would still leave a comment like @poire-z suggested, though.

FWIW, I feel like the most self-explanatory approach is the one in your latest quote (and that's despite me not being a fan of that sort of syntax and its heavy reliance on lexical scoping ;p).